Objective-C/Tests/DocumentExpirationTest.m

```diff
@@ -72,28 +72,38 @@ - (void) testSetExpirationToNonExistingDocument {
 - (void) testDocumentPurgedAfterExpiration {
     XCTestExpectation* expectation = [self expectationWithDescription: @"Document expiry test"];
     
-    // Create doc
+    NSError* err;
     CBLDocument* doc = [self generateDocumentWithID: nil];
-    AssertNil([self.db getDocumentExpirationWithID: doc.id]);
+    CBLMutableDocument* docToPurgeBeforeExpire = [self createDocument];
+    [docToPurgeBeforeExpire setValue: @"somevalue" forKey: @"somekey"];
+    [self saveDocument: docToPurgeBeforeExpire];
+    AssertEqual(self.db.count, 2u);
     
     // Setup document change notification
     id token = [self.db addDocumentChangeListenerWithID: doc.id
                                                listener: ^(CBLDocumentChange *change)
-    {
-        AssertEqualObjects(change.documentID, doc.id);
-        if ([change.database documentWithID: change.documentID] == nil) {
-            [expectation fulfill];
-        }
-        
-    }];
-    
+                {
+                    AssertEqualObjects(change.documentID, doc.id);
+                    if ([change.database documentWithID: change.documentID] == nil) {
+                        [expectation fulfill];
+                    }
+                }];
+
     // Set expiry
-    NSDate* expiryDate = [NSDate dateWithTimeIntervalSinceNow: 1.0];
-    NSError* err;
+    NSDate* expiryDateToPurge = [NSDate dateWithTimeIntervalSinceNow: 2.0];
+    Assert([self.db setDocumentExpirationWithID: docToPurgeBeforeExpire.id
+                                     expiration: expiryDateToPurge
+                                          error: &err]);
+    AssertNil(err);
+    NSDate* expiryDate = [NSDate dateWithTimeIntervalSinceNow: 3.0];
     Assert([self.db setDocumentExpirationWithID: doc.id expiration: expiryDate error: &err]);
     AssertNil(err);
     
-    // Wait for result
+    // purge one document
+    Assert([self.db purgeDocument: docToPurgeBeforeExpire error: &err]);
```

Comment:
From the name of the test case which is `testDocumentPurgedAfterExpiration`, I understand that it tested whether the document is purged after it got expired? What do you intend to test here?

Reply:
I understand it now. I think it's better to create a new test case to test purge before expired.

Reply:
I am the one that suggested they be the same @pasin because otherwise the only "assertions" in the test will be "no crash"

Reply:
I think it's testing different thing. The second test is testing whether purging doc before expired is ok or not. I feel like it's better to have them in two different test cases. At the end, it's about preferences.